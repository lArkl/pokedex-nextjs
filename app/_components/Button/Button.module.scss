.container {
  margin: 0;
  padding: 0.4rem 1rem;
  border-radius: 0.2rem;
  color: white;
  border-style: solid;
}

.primary {
  background-color: var(--color-red-2);

  &:hover,
  &:active {
    background-color: var(--color-red-1);
  }
}

.secondary {
  background-color: var(--color-purple-4);

  &:hover,
  &:active {
    background-color: var(--color-purple-3);
  }
}

.small {
  padding: 0.2rem 0.8rem;
}

.medium {
  padding: 0.4rem 1rem;
}

.large {
  padding: 0.8rem 1.6rem;
}
