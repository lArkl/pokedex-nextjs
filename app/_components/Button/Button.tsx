import { ButtonHTMLAttributes, FC } from 'react'
import styles from './Button.module.scss'
import classNames from 'classnames'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary'
  size?: 'small' | 'medium' | 'large'
}

const Button: FC<ButtonProps> = ({ variant = 'primary', size = 'medium', ...props }) => {
  return (
    <button {...props} className={classNames([styles.container, styles[variant], styles[size], props.className])}>
      {props.children}
    </button>
  )
}

export default Button
