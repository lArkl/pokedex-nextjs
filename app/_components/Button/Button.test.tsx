import { fireEvent, render, screen } from '@testing-library/react'
import Button from './Button'
import { vi } from 'vitest'

describe('Button', () => {
  it('renders component', async () => {
    const { asFragment } = render(<Button />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('shows inner text', async () => {
    render(<Button>Press</Button>)

    const button = screen.getByRole('button')
    expect(button).toHaveTextContent('Press')
  })

  it('accepts a className', async () => {
    render(<Button className="class" />)

    const button = screen.getByRole('button')
    expect(button).toHaveClass('class')
  })

  it('should fire on click', async () => {
    const onClickMock = vi.fn()
    render(<Button onClick={onClickMock} />)

    const button = screen.getByRole('button')
    fireEvent.click(button)

    expect(onClickMock).toHaveBeenCalled()
  })
})
