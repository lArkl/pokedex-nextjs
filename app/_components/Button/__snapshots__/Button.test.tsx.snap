// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders component 1`] = `
<DocumentFragment>
  <button
    class="_container_f8f296 _primary_f8f296 _medium_f8f296"
  />
</DocumentFragment>
`;
